### PR TITLE
Add new /access/ API

### PR DESCRIPTION
**Pull request by mistydemeo**
This API allows retrieving Archivist's Toolkit or ArchivesSpace records, using the same instances configured for use with the upload scripts.

### CODE REVIEW

File: src/archivematicaCommon/lib/archivesspace/client.py [hunk 1]
```diff
@@ -345,17 +350,42 @@ def count_collections(self, search_pattern='', identifier=''):
...
+            for example, "text" will match "this title has this text in it".
+            If omitted, then all resources will be fetched.
+        :param string identifier: Restrict records to only those with this identifier.
+            This refers to the human-assigned record identifier, not the automatically generated internal ID.
+            This value can contain wildcards.
+        :param string accession: Restrict records to only those associated with this accession.
+            This should be provided using ArchivesSpace's internal format, for instance /repositories/2/accessions/1; it does not refer to human-assigned accession identifiers.
+            Use ArchivesSpaceClient.find_collections_by_accession to locate records via a human-assigned accession identifier.
+
+        :return list: A list containing every matched resource's ID.
```
Comment (Hwesta) [inline src/archivematicaCommon/lib/archivesspace/client.py]: Sphinx doesn't support :return type: unfortunately.  It does support :return: and :returntype: http://sphinx-doc.org/domains.html#info-field-lists

File: src/dashboard/src/components/access/views.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        records = client.find_collections_by_accession(accession, page=page, page_size=page_size)
+        return helpers.json_response(records)
+    except ArchivesSpaceError as e:
+        return helpers.json_response({"error": True, "message": str(e)})
+
+
+@_authenticate_to_archivesspace
+def record_children(client, request, system='', record_id=''):
+    if request.method == 'PUT':
+        pass
```
Comment (Hwesta) [inline src/dashboard/src/components/access/views.py:110]: Why explicitly accept but do nothing on PUT?

**Comment by Hwesta** [comment]
:+1:

File: src/dashboard/src/components/access/views.py (hunk 2 above)
Comment (mistydemeo) [inline src/dashboard/src/components/access/views.py:110]: It's just stubbed in right now; the intention is for this to allow new records to be added. Now that the edit form is in, I'm going to have a followup PR to add the real functionality.

File: src/archivematicaCommon/lib/archivesspace/client.py (hunk 1 above)
Comment (mistydemeo) [inline src/archivematicaCommon/lib/archivesspace/client.py]: I'll fix this, and add a new commit that changes the rest.

Comment (mistydemeo) [inline src/archivematicaCommon/lib/archivesspace/client.py]: Fixed!

**Pull request merged**